ddspipe_core/test/unittest/efficiency/FastPayloadPoolTest.cpp

```diff
@@ -320,6 +340,66 @@ TEST(FastPayloadPoolTest, release_payload_negative)
     ASSERT_THROW(pool.release_payload(payload), eprosima::utils::InconsistencyException);
 }
 
+/**
+ * Check that concurrent get and release payload operations are thread safe (verified when executed with TSAN).
+ *
+ * @note This test is actually not supposed to pass with the current implementation of \c FastPayloadPool , it is the
+ * user's responsability to provide thread safety for this particular case.
+ */
+/* TEST(FastPayloadPoolTest, concurrent_get_release)
+   {
+    // Repeat the test several times, as the detection of a data race is not deterministic
+    const unsigned int NUM_ITERATIONS = 50;
+
+    int i = 0;
+    while (i < NUM_ITERATIONS)
+    {
+        FastPayloadPool pool;
+        Payload payload;
+
+        ASSERT_TRUE(pool.get_payload(DEFAULT_SIZE, payload));
+
+        std::thread t1(test::get_release, std::ref(pool), std::ref(payload));
+
+        ASSERT_TRUE(pool.release_payload(payload));
+
+        t1.join();
+
+        i++;
+    }
+   } */
+
+/**
+ * Check that concurrent release payload operations are thread safe (verified when executed with TSAN).
```

Comment:
Please explain briefly what the test does; e.g. verify that releasing a payload from two different threads doesn't cause a data race.